src/pages/Topup.vue

```diff
@@ -84,11 +100,14 @@ export default {
       this.scannerStarted = true
       this.theScanner = QRCodeScanner
     },
-    applyGiftCard() {
+    applyGiftCardCode(payload) {
```

Comment:
bind click to this action : applyGiftCard is enought
no need to create applyGiftCardCode just to call applyGiftCard 
the button is only clickable when its not in disable state, and the disabled state is again bound to haveGiftCardCode